notify/incidentio/incidentio.go

```diff
@@ -0,0 +1,204 @@
+// Copyright 2025 Prometheus Team
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package incidentio
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log/slog"
+	"net/http"
+	"os"
+	"strings"
+
+	commoncfg "github.com/prometheus/common/config"
+
+	"github.com/prometheus/alertmanager/config"
+	"github.com/prometheus/alertmanager/notify"
+	"github.com/prometheus/alertmanager/template"
+	"github.com/prometheus/alertmanager/types"
+)
+
+// Notifier implements a Notifier for incident.io.
+type Notifier struct {
+	conf    *config.IncidentioConfig
+	tmpl    *template.Template
+	logger  *slog.Logger
+	client  *http.Client
+	retrier *notify.Retrier
+}
+
+// New returns a new incident.io notifier.
+func New(conf *config.IncidentioConfig, t *template.Template, l *slog.Logger, httpOpts ...commoncfg.HTTPClientOption) (*Notifier, error) {
+	// If alert source token is specified, set authorization in HTTP config
+	if conf.HTTPConfig == nil {
+		conf.HTTPConfig = &commoncfg.HTTPClientConfig{}
+	}
+
+	if conf.AlertSourceToken != "" {
+		if conf.HTTPConfig.Authorization == nil {
```

Comment:
I think it's surprising that `alert_source_token` is discarded when `http_config` also contains authorization credentials. This feels like a case that should fail with an error?